zips/zip-tbd1.md

```diff
@@ -0,0 +1,129 @@
+    ZIP: Unassigned
+    Title: Media Memos Protocol (MMP)
+    Owners: Kyle Den Hartog <kdenhartog@brave.com>
+    Status: Draft
+    Category: Wallet
+    Created: 2024-08-27
+    License: MPL 2.0
+    Pull-Request: <https://github.com/zcash/zips/pull/899>
+
+
+# Abstract
+
+In this specification we’ll define how to include an on-chain pointer to a off-chain rich media message which is larger than 512 UTF-8 characters while maintaining the confidentiality guarantees of the memos defined in [^ZIP-302].
+
+# Motivation
+
+Today it's not possible to send a message larger than 512 UTF-8 characters in a memo. In this ZIP we'll define how this can be done while still maintaining confidentiality between the two communicating parties so that larger messages can be sent and received without needing to store all data on chain. This will allow for use cases like including a PDF invoice with a transaction or sending large personalized messages or videos directly to a ZCash address.
+
+# Terminology
+
+The key words "MUST", "MUST NOT", "SHOULD", "SHOULD NOT", "MAY",
+"RECOMMENDED", "OPTIONAL", and "REQUIRED" in this document are to
+be interpreted as described in BCP 14 [^BCP-14]_ when, and only when,
+they appear in all capitals.
+
+Definitions of commonly used terms used throughout this ZIP:
+ - Scheme - This is the scheme of the URI necessary to identify that this is a MMP message. It SHALL match the definition of a scheme as defined in [^RFC-3986] section 3.1.
+ - Version - This is the version of the Media Memo Protocol in use in order to support extensions of the protocol.
+ - Message Location - A URL (or derived URL as defined by a separate version spec) which can be used to locate the ciphertext of the message.
+ - Query Parameters - This is an extensible model for including additional query parameters directly within the MMP message URI. It SHALL match the definition of a query as defined by [^RFC-3986] section 3.4.
+
+# Requirements
+
+This spec should allow ZCash users to include rich media messages in their transactions through the use of memos as defined in RFC-302. Additionally, in order to maintain cryptographic agility and account for different means of sending messages, it should be extensible by design while still maintaining interoperability.
+
+# Specification
+
+## Message URI syntax
+
+The current memo zip requires that the content of the message is less than 512 ASCII characters which is a sufficient size to include a URI which includes the sufficient details to locate and decrypt the message. Here is how the format of the pointer will be:
+
+## ABNF definition of message URI syntax
+    mmp-uri                     = mmp ":" version ":" cid query-params
+    mmp-uri                     = /1*512VCHAR; Limit to 512 ASCII VCHARs
```

Comment:
To avoid parsing issues with URI libraries, it does make sense to keep this URI safe. However, because this needs maximum extensibility such that a version has the versatility it might need with various reserve characters I'm thinking it makes sense to make this `unreserved` / `reserved` which I think would be correct here in saying that any URI safe character can be used as long as the total length is less than 512 characters.

Does that sound good to you?

Reply:
The important thing is that parsing ambiguities aren't introduced here. Also, since this isn't valid ABNF (to have two productions with different constraints) there should be some other mechanism used to express this constraint.

Reply:
> The important thing is that parsing ambiguities aren't introduced here.

Agreed, will make sure that this is complying with the URI spec.

> Also, since this isn't valid ABNF (to have two productions with different constraints) there should be some other mechanism used to express this constraint.

I believe this would be valid ABNF under: https://datatracker.ietf.org/doc/html/rfc5234#section-3.3